nexus/db-queries/src/db/datastore/physical_disk.rs

```diff
@@ -34,42 +38,30 @@ impl DataStore {
     /// Stores a new physical disk in the database.
     ///
     /// - If the Vendor, Serial, and Model fields are the same as an existing
-    /// row in the table, the following fields may be updated:
-    ///   - Sled ID
-    ///   - Time Deleted
-    ///   - Time Modified
+    /// row in the table, an error is thrown.
     /// - If the primary key (ID) is the same as an existing row in the table,
     /// an error is thrown.
-    pub async fn physical_disk_upsert(
+    pub async fn physical_disk_insert(
         &self,
         opctx: &OpContext,
         disk: PhysicalDisk,
     ) -> CreateResult<PhysicalDisk> {
         let conn = &*self.pool_connection_authorized(&opctx).await?;
-        Self::physical_disk_upsert_on_connection(&conn, opctx, disk).await
+        Self::physical_disk_insert_on_connection(&conn, opctx, disk).await
     }
 
-    pub async fn physical_disk_upsert_on_connection(
+    pub async fn physical_disk_insert_on_connection(
         conn: &async_bb8_diesel::Connection<db::DbConnection>,
         opctx: &OpContext,
         disk: PhysicalDisk,
     ) -> CreateResult<PhysicalDisk> {
         opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
         use db::schema::physical_disk::dsl;
 
-        let now = Utc::now();
         let sled_id = disk.sled_id;
         let disk_in_db = Sled::insert_resource(
             sled_id,
-            diesel::insert_into(dsl::physical_disk)
-                .values(disk.clone())
-                .on_conflict(on_constraint("vendor_serial_model_unique"))
-                .do_update()
-                .set((
-                    dsl::sled_id.eq(excluded(dsl::sled_id)),
-                    dsl::time_deleted.eq(Option::<DateTime<Utc>>::None),
-                    dsl::time_modified.eq(now),
-                )),
+            diesel::insert_into(dsl::physical_disk).values(disk.clone()),
         )
         .insert_and_get_result_async(conn)
         .await
```

Comment:
Should the `ErrorHandler` in the .map_err after this line use `ErrorHandler::Conflict` now that this is an insert instead of an upsert?